phpBB/phpbb/install/controller/update.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\install\controller;
+
+use phpbb\install\helper\config;
+use phpbb\install\helper\install_helper;
+use phpbb\install\helper\iohandler\factory;
+use phpbb\install\helper\navigation\navigation_provider;
+use phpbb\language\language;
+use phpbb\request\request_interface;
+use phpbb\template\template;
+
+/**
+ * Updater controller
+ */
+class update
+{
+	/**
+	 * @var helper
+	 */
+	protected $controller_helper;
+
+	/**
+	 * @var install_helper
+	 */
+	protected $install_helper;
+
+	/**
+	 * @var factory
+	 */
+	protected $iohandler_factory;
+
+	/**
+	 * @var language
+	 */
+	protected $language;
+
+	/**
+	 * @var navigation_provider
+	 */
+	protected $menu_provider;
+
+	/**
+	 * @var request_interface
+	 */
+	protected $request;
+
+	/**
+	 * @var template
+	 */
+	protected $template;
+
+	// @todo: docblock
+	public function __construct(helper $controller_helper,
+								install_helper $install_helper,
+								factory $iohandler,
+								language $language,
+								navigation_provider $menu_provider,
+								request_interface $request,
+								template $template)
+	{
+		$this->controller_helper	= $controller_helper;
+		$this->install_helper		= $install_helper;
+		$this->iohandler_factory	= $iohandler;
+		$this->language				= $language;
+		$this->menu_provider		= $menu_provider;
+		$this->request				= $request;
+		$this->template				= $template;
+	}
+
+	/**
+	 * Controller entry point
+	 */
+	public function handle()
+	{
+		if (!$this->install_helper->is_phpbb_installed())
+		{
+			die ('phpBB is not installed');
```

Comment:
I thought redirection instead all die commands, as it is not requiring any phpBB code.